File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1847.lua
function split(inputstr, sep)
    if sep == nil then
            sep = "%s"
    end
    local t={}
    for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
            table.insert(t, str)
    end
    return t
end

linha = io.read()
ls = split(linha)

a = tonumber(ls[1])
b = tonumber(ls[2])
c = tonumber(ls[3])

if(a>b and b<=c) then feliz=true
elseif(b>a and c>b and c-b>=b-a) then feliz=true
elseif(a>b and b>c and b-c<a-b) then feliz=true
elseif(a==b and c>b) then feliz=true end
        
if(feliz) then print(":)")
else print(":(") end

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1847.cs
using System;

class URI
{
    static void Main() {

        string linha = Console.ReadLine();
        
        string[] ls = linha.Split(' ');
        
        int a = int.Parse(ls[0]), b = int.Parse(ls[1]), c = int.Parse(ls[2]); 
        
        bool feliz=false;
        
        if(a>b && b<=c) feliz=true;
        else if(b>a && c>b && c-b>=b-a) feliz=true;
        else if(a>b && b>c && b-c<a-b) feliz=true;
        else if(a==b && c>b) feliz=true;
        
        if(feliz) Console.Write(":)\n");
        else Console.Write(":(\n");
    }
}